skymap_scanner/server/start_scan.py

```diff
@@ -419,30 +466,54 @@ async def collect(
             if best.nside not in self.nsides_dict:
                 self.nsides_dict[best.nside] = {}
             if best.pixel in self.nsides_dict[best.nside]:
-                raise DuplicatePixelRecoException(
+                raise ExtraPixelRecoException(
                     f"NSide {best.nside} / Pixel {best.pixel} is already in nsides_dict"
                 )
             self.nsides_dict[best.nside][best.pixel] = best
             LOGGER.debug(f"Saved (found during {logging_id}): {best.id_tuple} {best}")
 
         # report after potential save
-        await self.reporter.record_pixreco(pixreco.nside, pixreco_start, pixreco_end)
+        await self.reporter.record_pixreco(
+            pixreco.nside,
+            pixreco_runtime,
+            roundtrip_start=self._pixreco_ids_sent__with_times[pixreco.id_tuple],
+            roundtrip_end=time.time(),
+        )
+
+    def is_scan_done(self) -> bool:
+        """Has every pixel been collected?"""
+        return self.n_received == self.n_sent
```

Comment:
This is a change from the previous `if col.pixreco_ids_sent == col.pixreco_ids_received`. Does checking the lengths here guarantee the received set of pixrecos is equal to the sent? Perhaps better to compare the keys of the sent pixel dictionary to the received set?

Reply:
Good point, I've added back the check